lib/ts/recipe/emailverification/emailVerificationClaim.ts

```diff
@@ -21,13 +22,27 @@ export class EmailVerificationClaimClass extends BooleanClaim {
                 id: this.id,
                 refresh: this.refresh,
                 shouldRefresh: (payload, userContext) => {
+                    const DateProvider = DateProviderReference.getReferenceOrThrow().dateProvider;
+
+                    if (maxAgeInSeconds < DateProvider.getThresholdInSeconds()) {
+                        throw new Error(
+                            `maxAgeInSeconds must be greater than the DateProvider threshold value -> ${DateProvider.getThresholdInSeconds()}`
+                        );
+                    }
+
+                    if (refetchTimeOnFalseInSeconds < DateProvider.getThresholdInSeconds()) {
+                        throw new Error(
+                            `refetchTimeOnFalseInSeconds must be greater than the DateProvider threshold value -> ${DateProvider.getThresholdInSeconds()}`
+                        );
+                    }
```

Comment:
why are these in the shouldRefetch and not in the isVerified function? Cause in supertokens-website, we have these checks outside the shouldRefresh functions.

Reply:
Is it cause of the issue you mentioned on slack where we init the validator before the dateprovider is init? Can you show me where this is being init before?

ALso, if this is the only solution, should we change the supertokens-website repo to also have these checks in the shouldRefresh? Consistency is required.

Reply:
Good catch. I have fixed it.. Since this function was using implicit return statement, I missed that. Moved it back to the top of the validator.